### ISSUE TITLE
PasswordGrantCondition is broken

### ISSUE THREAD

**Issue by albrechtflo-hg** [micronaut-projects/micronaut-security#979]
### Expected Behavior

OAuth2 Password Grant should be usable e.g. for HTTP client connections.

### Actual Behaviour

OAuth2 Password Grant is not initialized properly in the application context.

This is most likely due to this commit:
https://github.com/micronaut-projects/micronaut-security/commit/663cb062549d536764570bb1857204ebfb75538e#diff-88fced055e131397c9a1d120c000f301af40cac7049c325b1f375011e62027dbR54

The references to `failureMsgPrefix` in `Qualifiers.byName()` looks totally weird, and according to my debugging, does indeed not find required beans, thus does not enable this mode.

I am not sure if this also affects similar classes, as I currently am only trying to get ROPC working for a Microsoft REST API.

### Steps To Reproduce

My example configuration in `application.yml`:

```yaml
micronaut:
  security:
    redirect:
      enabled: false
    intercept-url-map:
      -
        pattern: /
        http-method: GET
        access:
          - isAnonymous()
    oauth2:
      clients:
        microsoft:
          client-id: 'abc-def-ghi-jkl'
          client-secret: 'topsecret123'
          grant-type: password
          client-credentials:
            service-id-regex: 'microsoft'
          token:
            url: "https://login.microsoftonline.com/my-tenant-id/oauth2/v2.0/token"
          authorization:
            url: "https://login.microsoftonline.com/my-tenant-id/oauth2/v2.0/auth"
  http:
    services:
      microsoft:
        connect-timeout: 5s
```

### Version

3.4.1

**Comment by sdelamo** [comment]
@alvarosanchez can you investigate this?

**Comment by jeremyg484** [comment]
This looks to have already been fixed in #1087
